tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,891 @@
+    WithTradingSessions,
+    ZiplineTestCase,
+)
+from zipline.testing.predicates import assert_equal
+from zipline.utils.numpy_utils import datetime64ns_dtype
+from zipline.utils.numpy_utils import float64_dtype
+
+
+class Estimates(DataSet):
+    event_date = Column(dtype=datetime64ns_dtype)
+    fiscal_quarter = Column(dtype=float64_dtype)
+    fiscal_year = Column(dtype=float64_dtype)
+    estimate = Column(dtype=float64_dtype)
+
+
+def QuartersEstimates(num_qtr):
+    class QtrEstimates(Estimates):
+        num_quarters = num_qtr
+        name = Estimates
+    return QtrEstimates
+
+
+class WithEstimates(WithTradingSessions, WithAssetFinder):
+    """
+    ZiplineTestCase mixin providing cls.loader and cls.events as class
+    level fixtures.
+
+
+    Methods
+    -------
+    make_loader() -> PipelineLoader
+        Method which returns the loader to be used throughout tests.
+    """
+
+    # Short window defined in order for test to run faster.
+    START_DATE = pd.Timestamp('2014-12-28')
+    END_DATE = pd.Timestamp('2015-02-03')
+
+    @classmethod
+    def make_loader(cls, events, columns):
+        return None
+
+    @classmethod
+    def init_class_fixtures(cls):
+        cls.sids = cls.events[SID_FIELD_NAME].unique()
+        cls.columns = {
+            Estimates.event_date: 'event_date',
+            Estimates.fiscal_quarter: 'fiscal_quarter',
+            Estimates.fiscal_year: 'fiscal_year',
+            Estimates.estimate: 'estimate'
+        }
+        cls.loader = cls.make_loader(cls.events, {column.name: val for
+                                                  column, val in
+                                                  cls.columns.items()})
+        cls.ASSET_FINDER_EQUITY_SIDS = list(
+            cls.events[SID_FIELD_NAME].unique()
+        )
+        cls.ASSET_FINDER_EQUITY_SYMBOLS = [
+            's' + str(n) for n in cls.ASSET_FINDER_EQUITY_SIDS
+        ]
+        super(WithEstimates, cls).init_class_fixtures()
+
+
+class WithWrongNumQuarters(WithEstimates):
+    """
+    ZiplineTestCase mixin providing cls.events as a class level fixture and
+    defining a test for all inheritors to use.
+
+    Attributes
+    ----------
+    events : pd.DataFrame
+        A simple DataFrame with columns needed for estimates and a single sid
+        and no other data.
+
+    Tests
+    ------
+    test_wrong_num_quarters_passed()
+        Tests that loading with an incorrect quarter number raises an error.
+    """
+    events = pd.DataFrame({SID_FIELD_NAME: 0},
+                          columns=[SID_FIELD_NAME,
+                                   TS_FIELD_NAME,
+                                   EVENT_DATE_FIELD_NAME,
+                                   FISCAL_QUARTER_FIELD_NAME,
+                                   FISCAL_YEAR_FIELD_NAME,
+                                   'estimate'],
+                          index=[0])
+
+    def test_wrong_num_quarters_passed(self):
+        with self.assertRaises(ValueError):
```

Comment:
can we lift all but the expected failure out of the `with self.assertRaises`? This is wrapping too much with a fairly general error.